server/repository/src/db_diesel/assets/asset_log_row.rs

```diff
@@ -1,36 +1,97 @@
 use super::asset_log_row::asset_log::dsl::*;
 
+use crate::EqualFilter;
 use crate::RepositoryError;
 use crate::StorageConnection;
 
 use chrono::NaiveDateTime;
 use diesel::prelude::*;
 
+use diesel_derive_enum::DbEnum;
+use serde::{Deserialize, Serialize};
+
 table! {
     asset_log (id) {
         id -> Text,
         asset_id -> Text,
         user_id -> Text,
-        status -> Nullable<Text>,
+        status -> Nullable<crate::db_diesel::assets::asset_log_row::StatusMapping>,
         comment -> Nullable<Text>,
         #[sql_name = "type"] type_ -> Nullable<Text>,
-        reason -> Nullable<Text>,
+        reason -> Nullable<crate::db_diesel::assets::asset_log_row::ReasonMapping>,
         log_datetime -> Timestamp,
     }
 }
 
+#[derive(DbEnum, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+#[DbValueStyle = "SCREAMING_SNAKE_CASE"]
+
+pub enum Reason {
```

Comment:
Similar to names, I think the naming is too generic please reference other 'status/type' enums